### PR TITLE
Remove netstandard1.x dependencies in source-build

### LINKED ISSUE #4482: Remove netstandard1.x assets from SBRP
See https://github.com/dotnet/arcade/issues/13315

Many projects in the source-build-reference-packages repository currently target netstandard1.x. Targeting netstandard1.x is now discouraged with the latest .NET SDK and we should remove those from SBRP.

This issue is blocked until all repositories that consume SBRP removed their netstandard1.x assets.

### PR DESCRIPTION
**Pull request by ViktorHofer**
## Bug

Fixes: https://github.com/dotnet/source-build/issues/4482

Regression? Last working version:

## Description

- Remove unnecessary netstandard1.x dependencies
- Make build task projects target net and netfx
  (NuGet.Build.Tasks.Pack and build/docs.proj)

Unrelated:
- Mark build.sh as executable (chmod +x)

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [ ] N/A

### CODE REVIEW

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj [hunk 1]
```diff
@@ -41,7 +41,7 @@
     <None Include="$(BuildCommonDirectory)NOTICES.txt" Pack="true" PackagePath="\" />
   </ItemGroup>
 
-  <ItemGroup Condition=" '$(TargetFramework)' != '$(NETFXTargetFramework)' ">
+  <ItemGroup Condition=" '$(TargetFramework)' == '$(NETCoreTargetFramework)' ">
```
Comment (akoeplinger) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj:44]: use TFI instead?

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj [hunk 2]
```diff
@@ -41,7 +41,7 @@
...
     <PackageReference Include="Microsoft.Build.Framework" ExcludeAssets="runtime" GeneratePathProperty="true" />
```
Comment (ViktorHofer) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj:45]: @rainersigwald is it recommended to reference the MSBuild packages on all frameworks or just on the ones where references aren't available? Asking as in L35-L36, msbuild references are used from the .NET Framework installation when targeting netfx.

Comment (rainersigwald) [reply src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj:45]: The .NET Framework references are the "most compatible" option, but . . . do we really support modern Pack-in-MSBuild in Visual Studio 2010? That doesn't seem required.

I would lean toward "reference the packages at the explicit lowest-supported-version for the netfx output".

**Comment by ViktorHofer** [comment]
I will get back to this when I return from my leave but I would be happy if someone else wants to pick this up. cc @MichaelSimons

**Pull request closed without merging**